app/src/main/java/ch/epfl/sweng/studdybuddy/Fragments/SettingsFragment.java

```diff
@@ -94,20 +103,27 @@ void setUpLang() {
 
         FirebaseReference ref = new FirebaseReference();
 
-        ref.select(Messages.FirebaseNode.USERS).select(uId).get(User.class, new Consumer<User>() {
-            @Override
-            public void accept(User user) {
-                String selectedLanguage = user.getFavoriteLanguage() != null ? user.getFavoriteLanguage() : Language.EN;
-                spinnerLang.setSelection(Language.LanguageToInt(selectedLanguage));
-            }
-        });
 
         //Language spinner
         ArrayAdapter<String> dataAdapterLanguages = new ArrayAdapter<>(this.getContext(), android.R.layout.simple_spinner_item, Language.languages);
         dataAdapterLanguages.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
         spinnerLang.setAdapter(dataAdapterLanguages);
+        //select from local db first
+
+        updateLanguage(user);
+        ref.select(Messages.FirebaseNode.USERS).select(uId).get(User.class, new Consumer<User>() {
+            @Override
+            public void accept(User user) {
+                updateLanguage(user);
+            }
+        });
```

Comment:
Should abstract away the Consumers from the activity

Reply:
Good remark, will be refactored as well